src/arch/xtensa/Kconfig

```diff
@@ -17,3 +17,9 @@ config WAKEUP_HOOK
 	  This config should be selected by other platform-level configs.
 	  Platforms that use it, have to implement hook function
 	  platform_interrupt_on_wakeup.
+
+config INLINE_FUNCTIONS
+        bool
+        default n
+        help
+          Indicate whether no_inine_functions are added into build option.
```

Comment:
typo: s/no_ininie/no_inline/

I think both the config option name and description should be improved. We are not enabling/disable inline functions here, but rather a specific compiler option "-finline-functions".

I'd suggest INLINE_CONSIDER_ALL_FUNCTIONS as the config option name and explain in description this is related to "-finline-functions" compiler option and whether it is enabled/disabled.

Reply:
"consider" sounds a bit like we'd be using an AI for this decision making and even then it would be an exaggeration :-) Maybe something along the lines "automatic inlining?" Or "compiler inline" or "compiler inline optimization" etc. Maybe indeed verbatim `COMPILER_INLINE_FUNCTIONS`.

Reply:
Lets just say exactly what this is. i.e.
This options add -no_inline_functions to the compiler command line. Some old compilers may not emit correct code
when enabled.